include/simdjson/generic/ondemand/json_type.h

```diff
@@ -115,6 +132,7 @@ struct number {
     double floating_point_number;
     int64_t signed_integer;
     uint64_t unsigned_integer;
+    bigint_t big_integer;
```

Comment:
The union is meant to span 64-bit. A `bigint_t` is not a 64-bit data structure.

It is also very much unlike the other values because it requires a variable-length amount of memory (possibly on the string buffer).

Reply:
I see. The 64bit union size restriction wasn't obvious by reading the code, which is why I thought that it didn't matter.  Where is it in the code that expects the size of the number struct to be 64 bits?

Reply:
> Where is it in the code that expects the size of the number struct to be 64 bits?

The union is at least as big as necessary to hold its largest data member. Currently, all values use 64 bits so there is no waste. By introducing a type that is almost surely larger (I expect an std::string_view to span 16 bytes typically), the number type is thus made larger.

How this affects the code and its performance is hard to tell in the abstract, but we cannot assume that it is free. It could be free, but we have to prove that it is so.

We should try to optimize for the common case. And the common case is that there is no big integer. We do not want everyone to be paying the price for the uncommon cases when there are big integers... 

Now, it is possible that your design is correct and that there is no trade-off but my instinct is to make the big integer case exceptional.

In any case, we have to remember that the big integer type is not usable as is. That is, you get a string-view. You are still going to need to parse it into a usable type. So it is bound to be a special case.


In any case, we need to justify the design and make sure we don't make everyone's code slower for a feature that very few people will ever use. That is no meant to be discouraging... for some people, your implementation might be absolutely super useful... but we don't want to trade that for slower performance everywhere else.